chains/solana/contracts/tests/ccip/ccip_router_test.go

```diff
@@ -2914,6 +2914,139 @@ func TestCCIPRouter(t *testing.T) {
 		})
 	})
 
+	///////////////////////////
+	// Withdraw billed funds //
+	////////////////////.......
+	t.Run("Withdraw billed funds", func(t *testing.T) {
+		t.Run("Preconditions", func(t *testing.T) {
+			require.Greater(t, getBalance(wsol.billingATA), uint64(0))
+			require.Greater(t, getBalance(token2022.billingATA), uint64(0))
+		})
+
+		t.Run("When an non-admin user tries to withdraw funds from a billing token account, it fails", func(t *testing.T) {
+			ix, err := ccip_router.NewWithdrawBilledFundsInstruction(
+				true,      // withdraw all
+				uint64(0), // amount
+				wsol.mint,
+				wsol.billingATA,
+				wsol.userATA, // send them to the user's account (just because, it could be any wallet)
+				wsol.program,
+				config.BillingSignerPDA,
+				config.RouterConfigPDA,
+				user.PublicKey(), // wrong user here
+			).ValidateAndBuild()
+			require.NoError(t, err)
+
+			testutils.SendAndFailWith(ctx, t, solanaGoClient, []solana.Instruction{ix}, user, config.DefaultCommitment, []string{ccip_router.Unauthorized_CcipRouterError.String()})
+		})
+
+		t.Run("When withdrawing funds but sending them to the wrong token account, it fails", func(t *testing.T) {
```

Comment:
what does it mean wrong in this context? A not-owned by admin account?

Reply:
The test is mixing tokens, trying to withdraw WSOL to a non-wsol token account. Regarding whitelisting of the target account, I'll create another PR for that (https://github.com/smartcontractkit/chainlink-ccip/pull/389#discussion_r1895916773)